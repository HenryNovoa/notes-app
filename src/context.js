import React from 'react'

export const NotesContext = React.createContext({
    currentNote: null,
    notes: [
        {id: 1, text: 'Do small assestment'},
        {id: 2, text: 'Go grocery shopping'}
    ]
})

