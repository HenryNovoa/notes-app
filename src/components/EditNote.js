import { useState, useContext, useRef, useEffect } from 'react'
import { NotesContext } from '../context'

export function EditNote() {
    const { state, dispatch } = useContext(NotesContext)
    const [ value, setValue ] = useState(state.currentNote.text)

    let ref = useRef()

    useEffect(()=> {
        ref.current.focus();
    }, [])

    const handleChange = (event) => {
        setValue(event.target.value)
    }

    const handleSubmit = (event) => {
        event.preventDefault()
        if(value.trim === '') return alert('Cannot add a blank note')

        dispatch({type: 'UPDATE_NOTE', payload: value})
        setValue('')
    }
    return (
        <div className='note-form'>
            <form onSubmit={handleSubmit}>
                <textarea cols='10' rows='10' type='text' ref={ref} onChange={handleChange} value={value}/>
                <button>Update Note</button>
            </form>
        </div>
    )
}