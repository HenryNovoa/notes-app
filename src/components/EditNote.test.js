import { shallow, mount } from 'enzyme'
import React, { useContext, useReducer } from 'react'
import { EditNote } from './EditNote'
import { NotesContext } from '../context'
import { notesReducer } from '../reducer'
import toJson from 'enzyme-to-json'



describe('EditNote Component', ()=> {
    it('expect to render EditNote component', ()=> {
       const wrapper = mount(<NotesContext.Provider value={{state: {
        currentNote: {id: 2, text: 'Write Tests'},
        notes: [
            {id: 1, text: 'Do small assestment'},
            {id: 2, text: 'Write Tests'}
        ]
    }}}>
           <EditNote/>
       </NotesContext.Provider>)

    const promise = () => {
    return new Promise(resolve => {
      setTimeout(() => {
        wrapper.update()
        resolve(wrapper)
      }, 3000)
    })
  }
   

    return promise().then((res) => {
        expect(toJson(res)).toMatchSnapshot()
      })
    })
   
})