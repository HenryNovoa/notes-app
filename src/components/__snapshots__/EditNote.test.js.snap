// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditNote Component expect to render EditNote component 1`] = `
<EditNote>
  <div
    className="note-form"
  >
    <form
      onSubmit={[Function]}
    >
      <textarea
        cols="10"
        onChange={[Function]}
        rows="10"
        type="text"
        value="Write Tests"
      />
      <button>
        Update Note
      </button>
    </form>
  </div>
</EditNote>
`;
